runtime/jvmti/jvmtiHook.c

```diff
@@ -969,6 +966,10 @@ jvmtiHookFieldAccess(J9HookInterface** hook, UDATA eventNum, void* eventData, vo
 					callback((jvmtiEnv *) j9env, (JNIEnv *) currentThread, threadRef, methodID, (jlocation) location, (jclass) classRef, (jobject) objectRef, fieldID);
 				}
 				finishedEvent(currentThread, JVMTI_EVENT_FIELD_ACCESS, hadVMAccess, javaOffloadOldState);
+				if (NULL != object) {
+					J9VMGetFieldEvent* data = eventData;
+					data->object = *objectRef;
```

Comment:
Shouldn't this be done under vmaccess?

Does it need to occur before finishedEvent?